### PR TITLE
Add RemoveTeamIcon plugin api

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
Add RemoveTeamIcon plugin api

#### Checklist
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: plugin/api.go [hunk 1]
```diff
@@ -74,6 +74,11 @@ type API interface {
 	// Minimum server version: 5.6
 	SetTeamIcon(teamId string, data []byte) *model.AppError
 
+	// RemoveTeamIcon remove the Team Icon.
```
Comment (hanzei) [inline plugin/api.go]:
```suggestion
	// RemoveTeamIcon removes the Team Icon.
```

Comment (cpanato) [reply plugin/api.go]: done

File: plugin/api.go [hunk 2]
```diff
@@ -74,6 +74,11 @@ type API interface {
 	// Minimum server version: 5.6
 	SetTeamIcon(teamId string, data []byte) *model.AppError
 
+	// RemoveTeamIcon removes the Team Icon.
```
Comment (jwilander) [inline plugin/api.go]:
```suggestion
	// RemoveTeamIcon removes the team icon.
```

Comment (cpanato) [reply plugin/api.go]: done

Comment (hanzei) [reply plugin/api.go]: I propose we also fix the other occurrences of this Term in https://github.com/mattermost/mattermost-server/blob/78cc31430fde7ed050fb0bba9a609f2c384445b5/plugin/api.go#L67 and
https://github.com/mattermost/mattermost-server/blob/78cc31430fde7ed050fb0bba9a609f2c384445b5/plugin/api.go#L72

Comment (cpanato) [reply plugin/api.go]: done @hanzei

**Comment by jwilander** [comment]
Needs a rebase @cpanato but then it's good to merge

**Pull request merged**